fdbserver/include/fdbserver/DDTeamCollection.h

```diff
@@ -311,7 +311,7 @@ class DDTeamCollection : public ReferenceCounted<DDTeamCollection> {
 
 	LocalityMap<UID> machineLocalityMap; // locality info of machines
 
-	Reference<KeyRangeMap<int>> customReplication;
+	std::shared_ptr<DDConfiguration::RangeConfigMapSnapshot> userRangeConfig;
```

Comment:
In the past there was a consensus `Reference` is preferred over `shared_ptr` unless we want to have some `weak_ptr` ability. Because Reference cost less for copy and storage. Any specific reason we use shared_ptr here?

Reply:
`Reference<>` requires the argument type to inherit `ReferenceCounted<>` so it is less generically useful.  If performance does not matter then there is no reason to use `Reference<>` over `std::shared_ptr<>`.  Reference is slightly less space overhead and guarantees the reference count is in the same allocation as the payload and by default it is thread unsafe, which are useful for performance but IMO when that doesn't matter it's awkward to inherit `ReferenceCounted<>` for no practical benefit.